modules/backend/serializer/SerializerHelper.hpp

```diff
@@ -120,9 +120,10 @@ ::clinic::staff::doctor::Doctor deserializeDoctor(const ::proto_staff::Doctor& p
 ::proto_staff::Staff serializeStaff(const ::clinic::staff::Staff& staff)
 {
     ::proto_staff::Staff proto_staff{};
-    for (const auto& doctor : staff.getDoctors())
+    for (const auto& doctor_ptr : staff.getDoctors())
     {
-        serializeDoctor(doctor, proto_staff.add_doctor());
+        const auto* doctor = dynamic_cast<const clinic::staff::doctor::Doctor*>(doctor_ptr);
+        serializeDoctor(*doctor, proto_staff.add_doctor());
```

Comment:
Instead dynamic_casting here, just modify serializeDoctor to take as an argument iDoctor and then reference to structures of passed ptr.

Reply:
Done - good idea